go-controller/pkg/ovn/namespace.go

```diff
@@ -235,9 +259,16 @@ func (oc *Controller) createNamespaceLocked(ns string) *namespaceInfo {
 	oc.namespacesMutex.Lock()
 	defer oc.namespacesMutex.Unlock()
 
+	namespace, err := oc.watchFactory.GetNamespace(ns)
+	if err != nil {
+		klog.Warningf("error getting namespace %s: %v", ns, err)
+	}
+	annotations := namespace.GetAnnotations()
+
 	nsInfo := &namespaceInfo{
 		addressSet:       make(map[string]string),
 		networkPolicies:  make(map[string]*namespacePolicy),
+		annotations:      annotations,
```

Comment:
you will set the annotation in the AddNamespace function, where you will already have the kapi.Namespace and you dont need to call watch factory